### PR TITLE
[SPARK-9355][SQL] Remove InternalRow.get generic getter call in columnar cache code

### PR DESCRIPTION
**Pull request by cloud-fan**

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @liancheng

**Comment by cloud-fan** [comment]
cc @rxin

**Comment by rxin** [comment]
@liancheng can you review this? Thanks.

**Comment by rxin** [comment]
OK I took a look and the changes looked alright. I'm going to merge this.

@liancheng it would be great if you can take a look at this post-hoc too.

File: sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnType.scala [hunk 1]
```diff
@@ -434,18 +441,18 @@ private[sql] object BINARY extends ByteArrayColumnType[BinaryType.type](11, 16)
 // Used to process generic objects (all types other than those listed above). Objects should be
 // serialized first before appending to the column `ByteBuffer`, and is also extracted as serialized
 // byte array.
-private[sql] object GENERIC extends ByteArrayColumnType[DataType](12, 16) {
+private[sql] case class GENERIC(dataType: DataType) extends ByteArrayColumnType(12, 16) {
   override def setField(row: MutableRow, ordinal: Int, value: Array[Byte]): Unit = {
-    row(ordinal) = SparkSqlSerializer.deserialize[Any](value)
+    row.update(ordinal, SparkSqlSerializer.deserialize[Any](value))
```
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnType.scala]: Not related to this PR, but it seems that we can replace `SparkSqlSerializer` with `SparkSqlSerializer2` here to gain better performance and small memory footprint here.

**Comment by liancheng** [comment]
Sorry, missed this PR yesterday. LGTM.

**Pull request closed without merging**